src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_validate.c

```diff
@@ -392,3 +392,23 @@ static int wm_agent_upgrade_compare_versions(const char *version1, const char *v
 
     return result;
 }
+
+
+bool wm_agent_upgrade_validate_task_update_message(const cJSON *response) {
```

Comment:
As I said, you should move this to the parser and make only one function with the request to the task module and the validation of the response like it is done in the function `wm_agent_upgrade_parse_task_module_task_ids`.